### PR TITLE
flyway migrationz

### PR DESCRIPTION
**Pull request by piercetrey-figure**
* Had to do a flyway baseline to exclude the uuid-ossp extension creation from non-postgres databases as this isn't supported
* Added regular flyway migration for creating tables previously created by SchemaUtils.create (if not exists)

### CODE REVIEW

File: server/src/main/kotlin/io/provenance/objectstore/gateway/configuration/DataConfig.kt [hunk 1]
```diff
@@ -52,31 +52,45 @@ class DataConfig {
...
+        .also {
+            TransactionManager.manager.defaultIsolationLevel = Connection.TRANSACTION_READ_COMMITTED
+            Database.registerDialect("pgsql") { PostgreSQLDialect() }
+            Database.registerDialect("sqlite") { SQLiteDialect() }
+            transaction {
+                if (databaseProperties.type == "postgresql") {
+                    try {
+                        SchemaUtils.createSchema(Schema(databaseProperties.schema))
+                    } catch (e: Exception) {
+                        logger.info("Exception creating schema [${databaseProperties.schema}]: ${e.message}")
```
Comment (hyperschwartz) [inline server/src/main/kotlin/io/provenance/objectstore/gateway/configuration/DataConfig.kt:67]: Nit: Should these be warns?  Or do they show up every time the app runs when the schema already exists?  If they always show up, infos are cool

File: migrations/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    id("org.flywaydb.flyway")
+}
+
+java.sourceCompatibility = JavaVersion.VERSION_11
+java.targetCompatibility = JavaVersion.VERSION_11
+
+val postgresHost = System.getenv("DB_HOST") ?: "127.0.0.1"
+val postgresPort = System.getenv("DB_PORT") ?: "5432"
+
+flyway {
```
Comment (hyperschwartz) [inline migrations/build.gradle.kts:18]: Do we need to actually include the flyway plugin configs in this repo?  Maybe just having the files is enough.  They might even be able to live in the server's resource folder instead?

**Review by hyperschwartz — APPROVED** [review]
APPORVED!  Address or ignore comments as you see fit lol

File: server/src/main/kotlin/io/provenance/objectstore/gateway/configuration/DataConfig.kt (hunk 1 above)
Comment (piercetrey-figure) [reply server/src/main/kotlin/io/provenance/objectstore/gateway/configuration/DataConfig.kt:67]: good call, I'll make them warns

File: migrations/build.gradle.kts (hunk 2 above)
Comment (piercetrey-figure) [reply migrations/build.gradle.kts:18]: that's a fair point, I guess if just running the app migrates stuff then we probably don't have a need to run it manually via gradle

**Pull request merged**